File: task1/main.rb
# Importing necessary libraries (none required in Ruby for this simple script)
print "Введите строку "  # Вывод приглашения для ввода строки
text = gets.chomp  

CountGlas = 0  # Счетчик гласных
CounterSog = 0  # Счетчик согласных

glas = "aeiouyаеёиоуыэюяAEIOUYАЕЁИОУЫЭЮЯ"  # Гласные буквы
soglas = "bcdfghjklmnpqrstvwxzбвгджзйклмнпрстфхцчшщBCDFGHJKLMNPQRSTVWXZБВГДЖЗЙКЛМНПРСТФХЦЧШЩ"  # Согласные буквы

# Согласные буквы
text.each_char do |i|
  if glas.include?(i)
    CountGlas += 1
  elsif soglas.include?(i)
    CounterSog += 1
  end
end

if CountGlas > CounterSog  # Сравнение количества гласных и согласных
  puts "да"
elsif CountGlas < CounterSog
  puts "нет"
else
  puts "одинаково"
end

File: task1/main.py
import string
text = input("Введите строку ") # Ввод строки

CountGlas=0 # Счетчик гласных
CounterSog=0 # Счетчик согласных

glas = "aeiouyаеёиоуыэюяAEIOUYАЕЁИОУЫЭЮЯ" # Гласные буквы
soglas = "bcdfghjklmnpqrstvwxzбвгджзйклмнпрстфхцчшщBCDFGHJKLMNPQRSTVWXZБВГДЖЗЙКЛМНПРСТФХЦЧШЩ" # Согласные буквы

# Согласные буквы
for i in text: 
    if i in glas:
        CountGlas += 1
    elif i in soglas:
        CounterSog += 1

if CountGlas > CounterSog: # Сравнение количества гласных и согласных
    print("да")
elif CountGlas < CounterSog:
    print("нет")
else:
    print("одинаково")
